File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/dto/entity/wordstatus/WordStatusResponseDTO.java
package ru.herooo.mylanguageweb.dto.entity.wordstatus;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import ru.herooo.mylanguageweb.controllers.json.LongDeserializer;
import ru.herooo.mylanguageweb.controllers.json.LongSerializer;
import ru.herooo.mylanguageweb.dto.entity.color.ColorResponseDTO;

public class WordStatusResponseDTO {
    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    private long id;

    @JsonProperty("title")
    private String title;

    @JsonProperty("code")
    private String code;

    @JsonProperty("message")
    private String message;

    @JsonProperty("color")
    private ColorResponseDTO color;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public ColorResponseDTO getColor() {
        return color;
    }

    public void setColor(ColorResponseDTO color) {
        this.color = color;
    }
}

File: MyLanguageWeb/src/main/resources/static/js/classes/dto/entity/word_status/response/word_status_response_dto.js
import {
    ColorResponseDTO
} from "../../color/response/color_response_dto.js";

export class WordStatusResponseDTO {
    #id;
    #title;
    #code;
    #description;
    #color;

    constructor(json) {
        if (json) {
            this.#id = json["id"];
            this.#title = json["title"];
            this.#code = json["code"];
            this.#description = json["description"];

            let color = json["color"];
            if (color) {
                this.#color = new ColorResponseDTO(color);
            }
        }
    }

    getId() {
        return this.#id;
    }

    getTitle() {
        return this.#title;
    }

    getCode() {
        return this.#code;
    }

    getDescription() {
        return this.#description;
    }

    getColor() {
        return this.#color;
    }
}